commons/src/main/java/com/ge/predix/acs/commons/web/BaseRestApiControllerAdvice.java

```diff
@@ -19,37 +19,118 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import org.springframework.beans.factory.annotation.Autowired;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.HttpMediaTypeNotAcceptableException;
+import org.springframework.web.HttpMediaTypeNotSupportedException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.json.MappingJackson2JsonView;
+
+import com.ge.predix.acs.commons.exception.UntrustedIssuerException;
 
 @ControllerAdvice
 public class BaseRestApiControllerAdvice {
 
-    @Autowired
-    private RestErrorHandler restErrorHandler;
-
-    /**
-     * Annotated method that handles a given exception.
-     *
-     * @param e        The given Exception
-     * @param request  The http request
-     * @param response The http response
-     * @return A model view containing a error response object
-     */
-    @ExceptionHandler(Exception.class)
-    public ModelAndView handleException(final Exception e, final HttpServletRequest request,
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseRestApiControllerAdvice.class);
+
+    private static void populateResponsesFromException(final RestApiException restApiException,
+            final HttpServletResponse httpServletResponse, final RestApiErrorResponse restApiErrorResponse) {
+        httpServletResponse.setStatus(restApiException.getHttpStatusCode().value());
+        restApiErrorResponse.setErrorMessage(restApiException.getMessage());
+        restApiErrorResponse.setErrorCode(restApiException.getAppErrorCode());
+    }
+
+    public RestApiErrorResponse preExceptionHandler(final Exception e, final HttpServletResponse response,
+            final int statusCode) {
+        LOGGER.error(e.getMessage(), e);
+        if (statusCode != 0) {
+            response.setStatus(statusCode);
+        } else {
+            response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+        }
+        return new RestApiErrorResponse();
+    }
+
+    @ExceptionHandler(RestApiException.class)
+    public ModelAndView handleRestApiException(final Exception e, final HttpServletRequest request,
+            final HttpServletResponse response) {
+        RestApiErrorResponse restApiErrorResponse = preExceptionHandler(e, response, 0);
+        populateResponsesFromException((RestApiException) e, response, restApiErrorResponse);
+        return new ModelAndView(new MappingJackson2JsonView(), "ErrorDetails", restApiErrorResponse);
```

Comment:
Spring now allows exception handlers to return ResposeEntity instead of ModelAndView.  The above method could be rewritten then:

    @ExceptionHandler
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    @ResponseBody
    RestApiErrorResponse handleRestApiException(final RestApiException e) {
        RestApiErrorResponse  restApiErrorResponse = new RestApiErrorResponse();
        restApiErrorResponse.setErrorMessage(e.getMessage());
        restApiErrorResponse.setErrorCode(e.getAppErrorCode());
        return restApiErrorResponse;
    }

Same could be applied to all exception handlers below.

Reply:
I have addressed all your review comments please review and let me know if something needs to changed